src/score/courses.py

```diff
@@ -83,61 +89,62 @@ def main(school: str, year: int, fields: str) -> None:
     # Drop courses for which the scoring field is empty
     courses_df = courses_df.dropna(subset=fields)
     # Concatenate the scoring fields
-    courses_df["text"] = courses_df[fields].apply(lambda x: "\n".join(x.values), axis=1)
-    courses_df = courses_df[["id", "languages", "text"]].set_index("id")
+    courses_df["scoring_text"] = courses_df[fields].apply(lambda x: "\n".join(x.values), axis=1)
+    courses_df["full_text"] = courses_df[["name", "content", "goal", "activity", "other"]]\
+        .apply(lambda x: "\n".join(x.values), axis=1)
+    courses_df = courses_df[["id", "languages", "scoring_text", "full_text"]].set_index("id")
 
     # Load patterns for different types of scores
-    themes_fn = Path(__file__).parent.absolute().joinpath("../../data/patterns/themes_patterns.json")
+    themes_fn = Path(__file__).parent.absolute().joinpath(f"../../data/patterns/{dictionary_name}.json")
     themes_patterns = json.load(open(themes_fn, 'r'))
     themes = themes_patterns.keys()
     patterns_matches_dict = {theme: {} for theme in themes}
-    for idx, text in courses_df.text.items():
-
-        # Clean text
-        text = text.lower()
-        chars_to_replace = ["\r", "\t", "\n", "\xa0", ":", ";", ".", ",", "?", "!", "(", ")", "…"]
-        for ch in chars_to_replace:
-            text = text.replace(ch, " ")
-
-        if len(text.strip(" ")) == 0:
-            courses_df.loc[idx, themes] = 0
+    scores_df = pd.DataFrame(0, index=courses_df.index, columns=themes, dtype=int)
+    for idx, (scoring_text, full_text) in courses_df[["scoring_text", "full_text"]].iterrows():
+
+        # Clean texts
+        def clan_text(text):
```

Comment:
Coquille dans le nom de la fonction : "clan_text" --> "clean_text".
Est-ce qu'il y a une raison précise pour laquelle tu génères la fonction on the fly à l'intérieur de la fonction parente `score_school_courses` plutôt qu'au dessus de cette dernière ? 
Ca me paraît une excellente idée de séparer les différentes étapes du traitement en plus petites fonctions pour améliorer la lisibilité du code. :)

Reply:
Généralement, je l'ai mets à l'intérieur de la fonciton parente quand ce n'est utilisé que par cette fonction. Par contre ici, c'est peut-être pas super malin de ma part de l'avoir mis dans la boucle...